driver/src/db_interface.rs

```diff
@@ -124,12 +124,21 @@ impl DbInterface for MongoDB {
         ];
 
         info!("Querying standing_orders: {:?}", pipeline);
-        self.client
+        let non_zero_standing_orders = self.client
             .db(models::DB_NAME)
             .collection("standing_orders")
             .aggregate(pipeline, None)?
             .map(|d| d.map(models::StandingOrder::from).map_err(DriverError::from))
-            .collect::<Result<Vec<_>, _>>()
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let empty_standing_order = models::StandingOrder::new(
+            0, 0, vec![]
+        );
+        let mut standing_orders = vec![empty_standing_order; models::NUM_RESERVED_ACCOUNTS as usize];
+        for standing_order in non_zero_standing_orders {
+            standing_orders[standing_order.account_id as usize] = standing_order.clone();
```

Comment:
I tried to be less clonely(), but did not find a way to do it. I have seen people switching to references or using std::mem::swap. But I think neither of these methods is appropriated here.

Reply:
Does foreach work (https://docs.rs/foreach/0.3.0/foreach/)?

> ForEach trait and for_each! macro allow you to use iterator inside iteration loop, which is not posible when using for-in loop.

Reply:
I use for_each now. It makes the code easier to understand. However, I was not able to remove the clone.